templates/ingress.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if .Values.ingress.enabled }}
+{{- range .Values.ingress.hosts }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "storageos.fullname" $ }}
+  namespace: {{ $.Values.namespace }}
+  labels:
+    app: {{ template "storageos.name" $ }}
+    chart: {{ template "storageos.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+  annotations:
+    {{- if .tls }}
+    ingress.kubernetes.io/secure-backends: "true"
+    {{- end }}
+    {{- range $key, $value := .annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+  - host: {{ .name }}
```

Comment:
Excuse me the ignorance, but where .name comes from? The host has to match exactly with the URL that the clients are going to come from. The header HTTP_HOST actually. Otherwise, the request will go to the default backend and 404.

Reply:
`.name` is from the elements in `.Values.ingress.hosts`. At the top of this file there's a `range` operator using which we loop through all the host configs. All the references inside this block are referred by their name directly. `.name` is the `name` attribute of a `hosts` entry. And we have leading `$` for some variables to get out of this loop scope and refer the from the global scope.